File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int maxLen = 0;
        int maxCount = 0;
        int i = 0;
        int n = s.length();
        Map<Character, Integer> map = new HashMap<>();
        for(int j=0;j<n;j++){
            map.put(s.charAt(j), map.getOrDefault(s.charAt(j), 0)+1);
            
            maxCount = Math.max(maxCount, map.get(s.charAt(j)));
            
            if(((j-i+1) - maxCount) > k){
                map.put(s.charAt(i), map.get(s.charAt(i))-1);
                i++;
            }
            
            maxLen = Math.max(maxLen, j-i+1);
        }
        return maxLen;
    }
}

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int maxLen = 0;
        int maxCount = 0;
        int n = s.size();
        int i = 0;
        unordered_map<char, int> map;
        for(int j = 0; j < n; j++){
            map[s[j]]++;
            
            maxCount = max(maxCount, map[s[j]]);
            
            if(((j-i+1)-maxCount) > k){
                map[s[i]]--;
                i++;
            }
            
            maxLen = max(maxLen, j-i+1);
        }
        return maxLen;
    }
};